Extensions/Leasing/Leasing.md

```diff
@@ -0,0 +1,54 @@
+
+[sf]: ../../Protocol.md#frame-setup
+
+### Metadata Contents
+```
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |L| Type ID/Len |   Lease Strategy                             ...
+    +---------------+---------------+---------------+---------------+
+    |L| Type ID/Len |   Lease Strategy                             ...
+    +---------------+---------------+---------------+---------------+
+                                   ...
+```
+
+* (**L**)Lease Strategy: Lease strategy is a well known value represented by a unique integer.  If L flag is set (a value of `1`), indicates a [Well-known Lease Strategy ID][wk].  If L flag is not set (a value of `0`), indicates the Lease Strategy Length in bytes.
+* **Lease Strategy ID/Length**: (7 bits = max value 2^7 = 128) Unsigned 7-bit integer.  If L flag is set (a value of `1`), indicates a [Well-known Lease Strategy ID][wk].  If A flag is not set (a value of `0`), indicates the Lease Strategy Length in bytes.
+* **Lease Strategy**: the strategy for leasing. This SHOULD be a US-ASCII string.  The string MUST NOT be null terminated.  (Not present if L flag is set)
+
+[wk]: WellKnownLeaseStrategies.md
+
+
+## Negotiation
+
+Once a `Client` specified [supported lease strategies](#metadata-payload), the `Server` MUST select one of the provided supported lease strategies or reject if none of the provided are supported.  In turn, to encode the selected lease strategy, `Server` MUST use the metadata field of the `LEASE`:
+
+### Metadata Contents
+```
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |L| Type ID/Len |   Lease Strategy                                  
+    +---------------+---------------+---------------+---------------+
+```
+
+* (**L**)Lease Strategy: Lease strategy is a well known value represented by a unique integer.  If L flag is set (a value of `1`), indicates a [Well-known Strategy Type ID][wk].  If L flag is not set (a value of `0`), indicates the Lease Strategy Length in bytes.
+* **Lease Strategy ID/Length**: (7 bits = max value 2^7 = 128) Unsigned 7-bit integer.  If L flag is set (a value of `1`), indicates a [Well-known Strategy Type ID][wk].  If A flag is not set (a value of `0`), indicates the Lease Strategy Length in bytes.
+* **Lease Strategy**: the strategy for leasing. This SHOULD be a US-ASCII string.  The string MUST NOT be null terminated.  (Not present if L flag is set)
+
+[wk]: WellKnownLeaseStrategies.md
+
+Note, if the selected lease strategy has a different frame layout than the default one, the first [`LEASE`][lf] frame MUST follow the default [`LEASE`][lf] frame layout, in order to let the `Client` read metadata field content 
+
```

Comment:
This should also deal with the possibility that a server does not specify any strategy in the `LEASE` frame such as when the server is unaware of and does not support the extension. I suppose in that case "Request Count" is chosen by default although the client may not have included that in its list of choices. Should "Request Count" be required then as a fallback?

Reply:
also, we should mentione that the first lease frame may have zero ttl and zero permits and amid to finalize lease handshake